internal/dag/dag.go

```diff
@@ -985,6 +985,9 @@ type Cluster struct {
 	TimeoutPolicy ClusterTimeoutPolicy
 
 	SlowStartConfig *SlowStartConfig
+
+	// MaxRequestsPerConnection defines the maximum number of requests per connection to the upstream before it is closed.
+	MaxRequestsPerConnection *uint32
```

Comment:
I don't think this is ever being set for Clusters? It'll need to be plumbed through to the HTTPProxy (at least) DAG processor so it can be set on the dag.Cluster, similar to DNSLookupFamily

Reply:
oof, fixed.